src/ripple/overlay/impl/TrafficCount.cpp

```diff
@@ -58,81 +76,122 @@ TrafficCount::category TrafficCount::categorize (
-                return (inbound && !msg->has_requestcookie()) ?
-                    TrafficCount::category::CT_get_trans :
-                    TrafficCount::category::CT_share_trans;
+        if (msg->type() == protocol::liTS_CANDIDATE)
             return (inbound && !msg->has_requestcookie()) ?
-                TrafficCount::category::CT_get_ledger :
-                TrafficCount::category::CT_share_ledger;
-        }
+                TrafficCount::category::ld_tsc_get:
+                TrafficCount::category::ld_tsc_share;
+
+        if (msg->type() == protocol::liTX_NODE)
+            return (inbound && !msg->has_requestcookie()) ?
+                TrafficCount::category::ld_txn_get :
+                TrafficCount::category::ld_txn_share;
+
+        if (msg->type() == protocol::liAS_NODE)
+            return (inbound && !msg->has_requestcookie()) ?
+                TrafficCount::category::ld_asn_get :
+                TrafficCount::category::ld_asn_share;
+
+        return (inbound && !msg->has_requestcookie()) ?
+               TrafficCount::category::ld_get :
+               TrafficCount::category::ld_share;
     }
 
+    if (auto msg = dynamic_cast<protocol::TMGetLedger const*>(&message))
     {
-        auto msg =
-            dynamic_cast <protocol::TMGetLedger const*>
-                (&message);
-        if (msg)
-        {
-            if (msg->itype() == protocol::liTS_CANDIDATE)
-                return (inbound || msg->has_requestcookie()) ?
-                    TrafficCount::category::CT_share_trans :
-                    TrafficCount::category::CT_get_trans;
+        if (msg->itype() == protocol::liTS_CANDIDATE)
+            return (inbound || msg->has_requestcookie()) ?
+                TrafficCount::category::gl_tsc_share :
+                TrafficCount::category::gl_tsc_get;
+
+        if (msg->itype() == protocol::liTX_NODE)
             return (inbound || msg->has_requestcookie()) ?
-                TrafficCount::category::CT_share_ledger :
-                TrafficCount::category::CT_get_ledger;
-        }
+                TrafficCount::category::gl_txn_share :
+                TrafficCount::category::gl_txn_get;
+
+        if (msg->itype() == protocol::liAS_NODE)
+            return (inbound || msg->has_requestcookie()) ?
+                TrafficCount::category::gl_asn_share :
+                TrafficCount::category::gl_asn_get;
+
+        return (inbound || msg->has_requestcookie()) ?
+            TrafficCount::category::gl_share :
+            TrafficCount::category::gl_get;
     }
 
+    if (auto msg = dynamic_cast<protocol::TMGetObjectByHash const*>(&message))
     {
-        auto msg =
-            dynamic_cast <protocol::TMGetObjectByHash const*>
-                (&message);
-        if (msg)
-        {
-            // inbound queries and outbound responses are sharing
-            // outbound queries and inbound responses are getting
+        if (msg->type() == protocol::TMGetObjectByHash::otLEDGER)
```

Comment:
Another way to go about this is pull these into a loop... makes it clear that the logic is identical for every case.

You could also cache the `bool const isInbound = msg->query() == inbound` outside the loop to avoid indirecting through the `msg` pointer for every check.

```
struct {
    message_type_enum message_type;
    TrafficCount::category share;
    TrafficCount::category get;
} checks[] = {
    { protocol::TMGetObjectByHash::otLEDGER,      TrafficCount::category::share_hash_ledger, TrafficCount::category::get_hash_ledger },
    { protocol::TMGetObjectByHash::otTRANSACTION, TrafficCount::category::share_hash_tx,     TrafficCount::category::get_hash_tx     },
    // etc. 
};

for (auto& check : checks) {
    if (msg->query() == check.message_type) {
        return (msg->query() == inbound) ? check.share : check.get;
    }
}
```

Reply:
Depending on the compiler, it might be able to consolidate those common expressions, but I would just do it myself as a matter of style. Gives an opportunity to give a meaningful name to the value, and makes it easier to repeat (via autocompletion) compared to the full expression.